src/services/codefixes/addNameToNamelessParameter.ts

```diff
@@ -26,24 +33,43 @@ registerCodeFix({
     getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) => makeChange(changes, diag.file, diag.start)),
 });
 
-function makeChange(changeTracker: textChanges.ChangeTracker, sourceFile: SourceFile, pos: number) {
-    const token = getTokenAtPosition(sourceFile, pos);
+function makeChange(changeTracker: textChanges.ChangeTracker, sourceFile: SourceFile, start: number) {
+    const token = getTokenAtPosition(sourceFile, start);
     const param = token.parent;
     if (!isParameter(param)) {
         return Debug.fail("Tried to add a parameter name to a non-parameter: " + Debug.formatSyntaxKind(token.kind));
     }
 
-    const i = param.parent.parameters.indexOf(param);
+    const parameters = param.parent.parameters;
+    const i = parameters.indexOf(param);
     Debug.assert(!param.type, "Tried to add a parameter name to a parameter that already had one.");
     Debug.assert(i > -1, "Parameter not found in parent parameter list.");
 
-    const typeNode = factory.createTypeReferenceNode(param.name as Identifier, /*typeArguments*/ undefined);
+    let end = param.name.getEnd();
+    let typeNode: TypeNode = factory.createTypeReferenceNode(param.name as Identifier, /*typeArguments*/ undefined);
+    let nextParam = tryGetNextParam(sourceFile, param);
+    while (nextParam) {
+        typeNode = factory.createArrayTypeNode(typeNode);
+        end = nextParam.getEnd();
+        nextParam = tryGetNextParam(sourceFile, nextParam);
+    }
+
     const replacement = factory.createParameterDeclaration(
         param.modifiers,
         param.dotDotDotToken,
         "arg" + i,
         param.questionToken,
-        param.dotDotDotToken ? factory.createArrayTypeNode(typeNode) : typeNode,
-        param.initializer);
-    changeTracker.replaceNode(sourceFile, param, replacement);
+        param.dotDotDotToken ? isArrayTypeNode(typeNode) ? typeNode : factory.createArrayTypeNode(typeNode) : typeNode,
+        param.initializer,
+    );
+    changeTracker.replaceRange(sourceFile, createRange(param.getStart(sourceFile), end), replacement);
+}
+
+function tryGetNextParam(sourceFile: SourceFile, param: ParameterDeclaration) {
+    const nextToken = findNextToken(param.name, param.parent, sourceFile);
+    if (nextToken && nextToken.kind === SyntaxKind.OpenBracketToken
+        && isArrayBindingPattern(nextToken.parent) && isParameter(nextToken.parent.parent)) {

```

Comment:
I think you want to validate that there are no bound names inside of this thing.

Reply:
Could you provide a sample?

Reply:
Sure, something like

```ts
(Something[a]) => string
```

Reply:
@DanielRosenwasser This case doesn't trigger QF, `checker` handles only identifiers. We could also think about change that...

```ts
if (isIdentifier(param.name)) {
  const originalKeywordKind = identifierToKeywordKind(param.name);
  if (
    (isCallSignatureDeclaration(param.parent) ||
      isMethodSignature(param.parent) ||
      isFunctionTypeNode(param.parent)) &&
    param.parent.parameters.indexOf(param) > -1 &&
    (resolveName(
      param,
      param.name.escapedText,
      SymbolFlags.Type,
      /*nameNotFoundMessage*/ undefined,
      param.name.escapedText,
      /*isUse*/ true,
    ) ||
      (originalKeywordKind && isTypeNodeKind(originalKeywordKind)))
  ) {
    const newName = "arg" + param.parent.parameters.indexOf(param);
    const typeName =
      declarationNameToString(param.name) + (param.dotDotDotToken ? "[]" : "");
    errorOrSuggestion(
      noImplicitAny,
      declaration,
      Diagnostics.Parameter_has_a_name_but_no_type_Did_you_mean_0_Colon_1,
      newName,
      typeName,
    );
    return;
  }
}
```

Reply:
@DanielRosenwasser Should I expand it to add support for `ElementAccessExpression`?